Code/Framework/AzFramework/AzFramework/Components/TransformComponent.cpp

```diff
@@ -624,10 +629,21 @@ namespace AzFramework
         // Ignore the event until we've already derived our local transform.
         if (m_parentTM)
         {
-            m_worldTM = parentWorldTM * m_localTM;
-            AZ::TransformNotificationBus::Event(
-                m_notificationBus, &AZ::TransformNotificationBus::Events::OnTransformChanged, m_localTM, m_worldTM);
-            m_transformChangedEvent.Signal(m_localTM, m_worldTM);
+            if (m_onParentChangedBehavior == AZ::OnParentChangedBehavior::Update)
+            {
+                m_worldTM = parentWorldTM * m_localTM;
+                AZ::TransformNotificationBus::Event(
+                    m_notificationBus, &AZ::TransformNotificationBus::Events::OnTransformChanged, m_localTM, m_worldTM);
+                m_transformChangedEvent.Signal(m_localTM, m_worldTM);
+            }
+            else
+            {
+                // Update the local transform to make sure it is consistent with the parent's new transform
+                // but keeps our world transform unchanged. Do not send any notifications here, because our world
+                // transform has not changed, and with this OnParentChangedBehavior setting the expectation is that
+                // another system will update our transform, and the notification will be triggered then.
+                m_localTM = parentWorldTM.GetInverse() * m_worldTM;
```

Comment:
When another system (rigid body for now) will update the transform, is it updating both the world and the local transforms? or is the local transform getting "out of sync" with the parent?

Reply:
The point is that the world transform will not change, but the local would (because the parent has moved relative to the child).

Reply:
>When physics (or whatever) calls SetWorldTransform, that will update both the world and child transform, so they won't get out of sync.

Perfect, thanks for the clarification!